site/content/api/Tenant-API.md

```diff
@@ -0,0 +1,264 @@
+
+1. Client has established an AMQP connection with Hono.
+2. Client has established an AMQP link in role *sender* with Hono using target address `tenant`. This link is used by the client to send tenant commands to Hono.
+3. Client has established an AMQP link in role *receiver* with Hono using source address `tenant/${reply-to}` where *reply-to* may be any arbitrary string chosen by the client. This link is used by the client to receive responses to the tenant api requests it has sent to Hono. This link's source address is also referred to as the *reply-to* address for the request messages.
+
+This flow of messages is illustrated by the following sequence diagram (showing the AMQP performatives):
+
+![Tenant message flow preconditions](../tenant_ConnectToTenant.png)
+
+# Operations
+
+The operations described in the following sections can be used by clients to manage tenant information. In real world scenarios the provisioning of tenants will most likely be an orchestrated process spanning multiple components of which Hono will only be one.
+
+Conducting and orchestrating the overall provisioning process is not in scope of Hono. However, Hono's Tenant API can easily be used as part of such an overall provisioning process.
+
+
+## Add Tenant
+
+Clients use this command to initially *add* information about a new tenant to Hono.
+
+This operation is *optional*, implementors of this API may provide other means for adding tenant information, e.g. a RESTful API or a configuration file.
+
+**Message Flow**
+
+The following sequence diagram illustrates the flow of messages involved in a *Client* creating a tenant.
+
+![Add tenant message flow](../tenant_AddTenantSuccess.png)
+
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on an *add tenant* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name        | Mandatory | Location                 | Type     | Description |
+| :---------- | :-------: | :----------------------- | :------- | :---------- |
+| *subject*   | yes       | *properties*             | *string* | MUST be set to `add`. |
+
+The request message MUST contain a tenant payload as defined in the [Tenant Format]({{< relref "#tenant-format" >}}) section below.
+
+In the payload it is also possible to put additional custom meta information about the tenant (see the [Tenant Format]({{< relref "#tenant-format" >}}) section for further information).
+ 
+**Response Message Format**
+
+A response to an *add tenant* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *201* | Created, the tenant has been successfully created. |
+| *409* | Conflict, there already exists a tenant for the given *tenant-id*. |
```

Comment:
should be `... given *tenant_id*`
NB: the tenant-id property should be renamed to tenant_id to support JMS based clients.

Reply:
adapted